src/platform/K32W/Logging.cpp

```diff
@@ -21,60 +21,54 @@
 static bool isLogInitialized;
 extern uint8_t gOtLogUartInstance;
 extern "C" void K32WWriteBlocking(const uint8_t * aBuf, uint32_t len);
+extern "C" uint32_t otPlatAlarmMilliGetNow(void);
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
-void GetMessageString(char * buf, uint8_t chipCategory, uint8_t otLevelLog)
+void GetMessageString(char * buf, uint8_t bufLen, const char * module, uint8_t category)
 {
-    if (chipCategory != kLogCategory_None)
+    int writtenLen = 0;
+    const char * categoryString;
+
+    writtenLen = snprintf(buf, bufLen, "[%lu]", otPlatAlarmMilliGetNow());
+    assert((writtenLen > 0) && (writtenLen < bufLen));
```

Comment:
Why is this assert valid?  What guarantees that `writtenLen >= bufLen` can't happen here?

Reply:
> Why is this assert valid? What guarantees that `writtenLen >= bufLen` can't happen here?

We can calculate the minimum size and document this as a precondition?